app/org/sagebionetworks/bridge/services/ConsentServiceImpl.java

```diff
@@ -42,189 +47,208 @@
     }
     @Autowired
-    public final void setActivityEventService(ActivityEventService activityEventService) {
+    final void setActivityEventService(ActivityEventService activityEventService) {
         this.activityEventService = activityEventService;
     }
+    @Autowired
+    final void setStudyEnrollmentService(StudyEnrollmentService studyEnrollmentService) {
+        this.studyEnrollmentService = studyEnrollmentService;
+    }
+    @Autowired
+    final void setSubpopulationService(SubpopulationService subpopService) {
+        this.subpopService = subpopService;
+    };
     
     @Override
-    public ConsentSignature getConsentSignature(final Study study, final User user) {
-        checkNotNull(user, Validate.CANNOT_BE_NULL, "user");
-        checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
+    public ConsentSignature getConsentSignature(Study study, SubpopulationGuid subpopGuid, User user) {
+        checkNotNull(study);
+        checkNotNull(subpopGuid);
+        checkNotNull(user);
+        
+        // This will throw an EntityNotFoundException if the subpopulation is not in the user's study
+        subpopService.getSubpopulation(study, subpopGuid);
         
         Account account = accountDao.getAccount(study, user.getEmail());
-        ConsentSignature signature = account.getActiveConsentSignature();
-        if (signature != null) {
-            return signature;
+        ConsentSignature signature = account.getActiveConsentSignature(subpopGuid);
+        if (signature == null) {
+            throw new EntityNotFoundException(ConsentSignature.class);    
         }
-        throw new EntityNotFoundException(ConsentSignature.class);
+        return signature;
     }
-
+    
     @Override
-    public User consentToResearch(final Study study, final User user, final ConsentSignature consentSignature,
-            final SharingScope sharingScope, final boolean sendEmail) {
-
+    public User consentToResearch(Study study, SubpopulationGuid subpopGuid, User user, ConsentSignature consentSignature, 
+            SharingScope sharingScope, boolean sendEmail) {
         checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
+        checkNotNull(subpopGuid, Validate.CANNOT_BE_NULL, "subpopulationGuid");
         checkNotNull(user, Validate.CANNOT_BE_NULL, "user");
         checkNotNull(consentSignature, Validate.CANNOT_BE_NULL, "consentSignature");
         checkNotNull(sharingScope, Validate.CANNOT_BE_NULL, "sharingScope");
 
-        if (user.doesConsent()) {
+        ConsentStatus status = user.getConsentStatuses().get(subpopGuid);
+        // There will be a status object for each subpopulation the user is mapped to. If 
+        // there's no status object, then in effect the subpopulation does not exist for 
+        // this user and they should get back a 404.
+        if (status == null) {
+            throw new EntityNotFoundException(Subpopulation.class);
+        }
```

Comment:
Added in the next commit: https://github.com/alxdarksage/BridgePF/commit/e12bbb1a0dd20043b63f646918e5cc1bf3ec85db